From 00ccec335792e3fa44712427463c64eb1ff9c5be Mon Sep 17 00:00:00 2001
From: progier389 <progier@redhat.com>
Date: Tue, 12 Jan 2021 17:45:41 +0100
Subject: [PATCH] Issue 4504 - insure that repl_monitor_test use ldapi (for
 RHEL) - fix merge issue (#4533)

(cherry picked from commit a880fddc192414d6283ea6832491b7349e5471dc)
---
 .../tests/suites/clu/repl_monitor_test.py     | 47 ++++++++++++++-----
 1 file changed, 36 insertions(+), 11 deletions(-)

diff --git a/dirsrvtests/tests/suites/clu/repl_monitor_test.py b/dirsrvtests/tests/suites/clu/repl_monitor_test.py
index b2cb840b3..caf6a9099 100644
--- a/dirsrvtests/tests/suites/clu/repl_monitor_test.py
+++ b/dirsrvtests/tests/suites/clu/repl_monitor_test.py
@@ -9,6 +9,7 @@
 import time
 import subprocess
 import pytest
+import re
 
 from lib389.cli_conf.replication import get_repl_monitor_info
 from lib389.tasks import *
@@ -69,6 +70,25 @@ def check_value_in_log_and_reset(content_list, second_list=None, single_value=No
         log.info('Reset log file')
         f.truncate(0)
 
+def get_hostnames_from_log(port1, port2):
+    # Get the supplier host names as displayed in replication monitor output
+    with open(LOG_FILE, 'r') as logfile:
+        logtext = logfile.read()
+    # search for Supplier :hostname:port 
+    # and use \D to insure there is no more number is after
+    # the matched port (i.e that 10 is not matching 101)
+    regexp = '(Supplier: )([^:]*)(:' + str(port1) + '\D)'
+    match=re.search(regexp, logtext)
+    host_m1 = 'localhost.localdomain'
+    if (match is not None):
+        host_m1 = match.group(2)
+    # Same for master 2 
+    regexp = '(Supplier: )([^:]*)(:' + str(port2) + '\D)'
+    match=re.search(regexp, logtext)
+    host_m2 = 'localhost.localdomain'
+    if (match is not None):
+        host_m2 = match.group(2)
+    return (host_m1, host_m2)
 
 @pytest.mark.ds50545
 @pytest.mark.bz1739718
@@ -177,20 +197,9 @@ def test_dsconf_replication_monitor(topology_m2, set_log_file):
                  '001',
                  m1.host + ':' + str(m1.port)]
 
-    dsrc_content = '[repl-monitor-connections]\n' \
-                   'connection1 = ' + m1.host + ':' + str(m1.port) + ':' + DN_DM + ':' + PW_DM + '\n' \
-                   'connection2 = ' + m2.host + ':' + str(m2.port) + ':' + DN_DM + ':' + PW_DM + '\n' \
-                   '\n' \
-                   '[repl-monitor-aliases]\n' \
-                   'M1 = ' + m1.host + ':' + str(m1.port) + '\n' \
-                   'M2 = ' + m2.host + ':' + str(m2.port)
-
     connections = [m1.host + ':' + str(m1.port) + ':' + DN_DM + ':' + PW_DM,
                    m2.host + ':' + str(m2.port) + ':' + DN_DM + ':' + PW_DM]
 
-    aliases = ['M1=' + m1.host + ':' + str(m1.port),
-               'M2=' + m2.host + ':' + str(m2.port)]
-
     args = FakeArgs()
     args.connections = connections
     args.aliases = None
@@ -198,8 +207,24 @@ def test_dsconf_replication_monitor(topology_m2, set_log_file):
 
     log.info('Run replication monitor with connections option')
     get_repl_monitor_info(m1, DEFAULT_SUFFIX, log, args)
+    (host_m1, host_m2) = get_hostnames_from_log(m1.port, m2.port)
     check_value_in_log_and_reset(content_list, connection_content, error_list=error_list)
 
+    # Prepare the data for next tests
+    aliases = ['M1=' + host_m1 + ':' + str(m1.port),
+               'M2=' + host_m2 + ':' + str(m2.port)]
+
+    alias_content = ['Supplier: M1 (' + host_m1 + ':' + str(m1.port) + ')',
+                     'Supplier: M2 (' + host_m2 + ':' + str(m2.port) + ')']
+
+    dsrc_content = '[repl-monitor-connections]\n' \
+                   'connection1 = ' + m1.host + ':' + str(m1.port) + ':' + DN_DM + ':' + PW_DM + '\n' \
+                   'connection2 = ' + m2.host + ':' + str(m2.port) + ':' + DN_DM + ':' + PW_DM + '\n' \
+                   '\n' \
+                   '[repl-monitor-aliases]\n' \
+                   'M1 = ' + host_m1 + ':' + str(m1.port) + '\n' \
+                   'M2 = ' + host_m2 + ':' + str(m2.port)
+
     log.info('Run replication monitor with aliases option')
     args.aliases = aliases
     get_repl_monitor_info(m1, DEFAULT_SUFFIX, log, args)
-- 
2.26.2

